enhancements/ingress/default-ingress-controller.md

```diff
@@ -0,0 +1,110 @@
+---
+title: default-ingress-controller
+authors:
+  - "@ironcladlou"
+reviewers:
+  - "@openshift/api-reviewers"
+  - "@openshift/sig-network-edge"
+  - "@ironcladlou"
+
+approvers:
+  - "@openshift/api-approvers"
+creation-date: 2020-04-06
+creation-date: 2020-04-06
+status: implementable
+see-also: [user-defined-default-ingress-controller](user-defined-default-ingress-controller.md)
+replaces:
+superseded-by:
+---
+
+# Default Ingress Controller Changes
+
+Today, unless the user explicitly [provides one at installation time](user-defined-default-ingress-controller.md), OpenShift's `default` IngressController resource is created by the ingress-operator. Importantly, ingress-operator will periodically check for the existence of the `default` instance, creating the resource if it does not exist. While this is useful to ensure some baseline ingress functionality within the cluster even in the event of the default resource being deleted, the current behavior is at odds with the common administrative practice of replacing the `default` instance as a day 2 operation. For example, to change the publishing strategy of the IngressController after installation, the resource must be replaced, causing a race between the administrator and the operator. This contention frustrates automation and can even cost resources if the operator wins (by creating unnecessary cloud resources like load balancers, public IPs, and DNS records.)
+
+This proposal eliminates the contention between the operator and administrators.
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+This proposal eliminates the operator/administrator contention by building on the [User Defined IngressControllers](user-defined-default-ingress-controller.md) capability. Specifically, by:
+
+1. Changing the ingress-operator to stop creating ingresscontrollers altogether
+2. Changing the installer to always render a `default` IngressController
```

Comment:
Currently the operator takes corrective action if there is no default ingresscontroller, and you are proposing that the operator instead leave it up to the user to take that corrective action.  A crucial part of the proposal is that the user must be cued to take this corrective action.  Therefore I believe the proposal should include the following:

```
3. Defining an alert that warns if there is no ingress controller with the cluster ingress domain, and defining an additional alert if there is no ingress controller at all.
```

Reply:
Agree, I would go even further and say that we should spell out how Degraded and Available should function in the absence of a `default` ingresscontroller

Reply:
And any other condition we want associated with the state

Reply:
Added a whole part about new conditions.